lib/astarte_rpc/config.ex

```diff
@@ -99,12 +132,45 @@ defmodule Astarte.RPC.Config do
     host = amqp_connection_host!()
     port = amqp_connection_port!()
 
-    [
+    options = [
       username: username,
       password: password,
       virtual_host: virtual_host,
-      host: host,
+      host: to_charlist(host),
       port: port
     ]
+
+    if amqp_connection_ssl_enabled!() do
+      ca_file =
+        case amqp_connection_ssl_ca_file() do
+          {:ok, nil} ->
+            CAStore.file_path()
+
+          {:ok, ca_cert_file} ->
+            ca_cert_file
+        end
+
+      ssl_options = [cacertfile: ca_file, verify: :verify_peer]
+
+      ssl_options =
+        if amqp_connection_ssl_disable_sni!() do
+          Keyword.put(ssl_options, :server_name_indication, :disable)
+        else
+          sni =
+            case amqp_connection_ssl_custom_sni() do
+              {:ok, nil} ->
+                amqp_connection_host!()
+
+              {:ok, sni} ->
+                sni
+            end
+
+          Keyword.put(ssl_options, :server_name_indication, to_charlist(sni))
+        end
+
+      Keyword.put(options, :ssl_options, ssl_options)
+    else
+      options
+    end
```

Comment:
Not fond of this code (usually if you see this `>` shape in code you should stop and think about what you're doing).
A cleaner way to handle this would be either using a `with` or breaking the code in subfunctions, e.g.
```elixir
...
  # end of options here
  ]
  |> populate_ssl_options()
...

defp populate_ssl_options(options) do
  if amqp_connection_ssl_enabled!() do
    ssl_options = build_ssl_options()
    Keyword.put(options, :ssl_options, ssl_options)
  else
    options
  end
end

defp build_ssl_options() do
  [
    cacertfile: amqp_connection_ssl_ca_file!() || CAStore.file_path(),
    verify: :verify_peer
  ]
  |> populate_sni()
end

defp populate_sni(ssl_options) do
  if amqp_connection_ssl_disable_sni!() do
    Keyword.put(ssl_options, :server_name_indication, :disable)
  else
    server_name =  amqp_connection_ssl_custom_sni!() || amqp_connection_host!()
    Keyword.put(ssl_options, :server_name_indication, to_charlist(server_name))
  end
end
```
The end result is the same but you break the logic in small pieces and you name every small piece so that you better communicate your intent to who is reading the code (i.e. the future you).